lib/plugins/aws/package/compile/events/msk/index.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+
+class AwsCompileMSKEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileMSKEvents.bind(this),
+    };
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'msk', {
+      type: 'object',
+      properties: {
+        arn: {
+          oneOf: [
+            {
+              type: 'string',
+              pattern:
+                '^arn:aws[a-zA-Z-]*:kafka:[a-z]{2}((-gov)|(-iso(b?)))?-[a-z]+-[1-9]{1}:[0-9]{12}:cluster',
+            },
+            { $ref: '#/definitions/awsCfImport' },
+            { $ref: '#/definitions/awsCfRef' },
+          ],
+        },
+        batchSize: {
+          type: 'number',
+          minimum: 1,
+          maximum: 10000,
+        },
+        enabled: {
+          type: 'boolean',
+        },
+        startingPosition: {
+          type: 'string',
+          enum: ['LATEST', 'TRIM_HORIZON'],
+        },
+        topic: {
+          type: 'string',
+        },
+      },
+      additionalProperties: false,
+      required: ['arn', 'topic'],
+    });
+  }
+
+  resolveDependsOn(funcRole) {
+    let dependsOn = 'IamRoleLambdaExecution';
+
+    if (funcRole) {
+      if (
+        // check whether the custom role is an ARN
+        typeof funcRole === 'string' &&
+        funcRole.indexOf(':') !== -1
+      ) {
+        dependsOn = [];
+      } else if (
+        // otherwise, check if we have an in-service reference to a role ARN
+        typeof funcRole === 'object' &&
+        'Fn::GetAtt' in funcRole &&
+        Array.isArray(funcRole['Fn::GetAtt']) &&
+        funcRole['Fn::GetAtt'].length === 2 &&
+        typeof funcRole['Fn::GetAtt'][0] === 'string' &&
+        typeof funcRole['Fn::GetAtt'][1] === 'string' &&
+        funcRole['Fn::GetAtt'][1] === 'Arn'
+      ) {
+        dependsOn = funcRole['Fn::GetAtt'][0];
+      } else if (
+        // otherwise, check if we have an import or parameters ref
+        typeof funcRole === 'object' &&
+        ('Fn::ImportValue' in funcRole || 'Ref' in funcRole)
+      ) {
+        dependsOn = [];
+      } else if (typeof funcRole === 'string') {
+        dependsOn = funcRole;
+      }
+    }
+
+    return dependsOn;
+  }
+
+  getMSKClusterName(eventSourceArn) {
```

Comment:
Let's maybe rename it to `getMSKClusterNameToken` (as we're not necessarily resolve the _cluster name_)